packages/bvaughn-architecture-demo/src/suspense/createGenericCache.ts

```diff
@@ -90,5 +98,26 @@ export function createGenericCache<TParams extends Array<any>, TValue>(
         }
       }
     },
+
+    useGetValue(...args: TParams) {
+      const [counter, setCounter] = useState(0);
+      function triggerRendering() {
+        setCounter(counter + 1);
+      }
+
+      const record = getOrCreateRecord(...args);
+      switch (record.status) {
+        case STATUS_PENDING: {
+          record.value.then(triggerRendering, triggerRendering);
```

Comment:
This looks side effecty. Rendering multiple times would end up attaching multiple `triggerRendering` callbacks.

Reply:
Yes, but they all resolve at the same time, so this should not trigger rerendering more than once.